### PR TITLE
[process-agent] Wrap Runner implementation into an fx Component

### PR DESCRIPTION
**Pull request by just-chillin**
### What does this PR do?

This PR refactors the `Runner` to be wrapped inside an `fx.Component`.

### Motivation

Migrating the process-agent to component architecture. 

#### Implementation Notes
At this point I decided to leave the `runner.MockModule` alone, since it requires the submitter to be implemented.
I also inject the raw `checks`, sysprobe config, and host info structs. Eventually all of these should be moved to components, but for now it's beyond the scope of this PR.

#### Changes Made to Other Packages
- `main` is changed to use `fxutil.App`, which lets FX start and stop the runner
- `pkg/process/runner` is changed such that the `runner` has a `Start` and `Stop` method, instead of an "exit" channel. This was done so that it was easier for fx to start and stop that component.
- `fxutil.fxLoggingOption` was made public.

### Describe how to test/QA your changes

- [x] Ensure that the process-agent starts and stops as expected. This means being able to start and stop the agent gracefully in container environments, as a service, and through the command line.
- [x] Ensure that the process-agent is still able to run checks and submit them to our intake.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [x] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [x] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: comp/process/runner/runner.go [hunk 1]
```diff
@@ -8,20 +8,24 @@ package runner
...
 
 	"go.uber.org/fx"
 
 	"github.com/DataDog/datadog-agent/comp/core/config"
 	"github.com/DataDog/datadog-agent/comp/core/sysprobeconfig"
 	"github.com/DataDog/datadog-agent/comp/process/submitter"
 	"github.com/DataDog/datadog-agent/comp/process/types"
+	ddconfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/process/checks"
+	r "github.com/DataDog/datadog-agent/pkg/process/runner"
```
Comment (just-chillin) [inline comp/process/runner/runner.go:20]: TODO: Need a better name for this.

File: comp/process/bundle_test.go [hunk 2]
```diff
@@ -6,32 +6,46 @@
...
+
+			[]checks.Check{checkMocks.NewCheck(t)},
+			&checks.HostInfo{},
+			&sysconfig.Config{},
+		),
 
 		// instantiate all of the process components, since this is not done
 		// automatically.
 		fx.Invoke(func(r runner.Component) {}),
-		fx.Invoke(func(r submitter.Component) {}),
```
Comment (just-chillin) [inline comp/process/bundle_test.go:30]: As a heads up, I will add invocations for these components as I build them.

For now, I want to be able to control the components that are constructed, as some of the downstream dependants of components in the process bundle that I haven't touched yet change global state when they are constructed.

File: comp/process/runner/runner.go (hunk 1 above)
Comment (kkhor-datadog) [reply comp/process/runner/runner.go:16]: I typically just append the higher-level folder name, so in this case it would be `process-runner`

Comment (just-chillin) [reply comp/process/runner/runner.go:16]: Thanks for the suggestion!

**Review by kkhor-datadog — APPROVED** [review]

File: comp/process/runner/runner_test.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	"go.uber.org/fx"
+
+	sysconfig "github.com/DataDog/datadog-agent/cmd/system-probe/config"
+	"github.com/DataDog/datadog-agent/pkg/process/checks"
+	checkMocks "github.com/DataDog/datadog-agent/pkg/process/checks/mocks"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+)
+
+func newMockCheck(t testing.TB, name string) *checkMocks.Check {
+	// TODO: Change this to use check componenet once checks are migrated
```
Comment (ogaca-dd) [inline comp/process/runner/runner_test.go:21]:
```suggestion
	// TODO: Change this to use check component once checks are migrated
```

File: cmd/process-agent/main_common.go [hunk 4]
```diff
@@ -258,31 +262,46 @@ func runAgent(globalParams *command.GlobalParams, exit chan struct{}) {
...
+		log.Info("The process-agent has successfully been shut down")
 	}
-
-	for range exit {
-	}
-
-	if err := srv.Shutdown(context.Background()); err != nil {
-		log.Errorf("Error shutting down expvar server on port %v: %v", expVarPort, err)
-	}
+	return
```
Comment (ogaca-dd) [inline cmd/process-agent/main_common.go:304]: `return` is not needed.

**Review by ogaca-dd — APPROVED** [review]
LGTM

File: cmd/process-agent/main_common.go (hunk 4 above)
Comment (just-chillin) [reply cmd/process-agent/main_common.go]: Thanks, updated!

**Pull request merged**